python/cudnn/sdpa/fwd/torch_op.py

```
@@ -0,0 +1,973 @@
+    if is_thd:
+        if cu_seqlens_kv is None or max_seqlen_q <= 0 or max_seqlen_kv <= 0:
+            raise ValueError("varlen path needs cu_seqlens_q, cu_seqlens_kv, max_seqlen_q, max_seqlen_kv")
+        if q.ndim != 3:
+            raise ValueError(f"varlen path expects packed (T, H, D) tensors, got q.ndim={q.ndim}")
+        if seq_len_q is not None or seq_len_kv is not None:
+            raise ValueError("varlen path derives seq lens from cu_seqlens; do not pass seq_len_q/kv")
+        B = cu_seqlens_q.numel() - 1
+        q = _normalize_thd(q, "q")
+        k = _normalize_thd(k, "k")
+        v = _normalize_thd(v, "v")
+        T_q, H_q, D_qk = q.shape
```

Ruff (0.16.2)

[warning] Unpacked variable `T_q` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)